src/jvmMain/kotlin/com/charleskorn/kaml/YamlOutput.kt

```diff
@@ -90,6 +92,23 @@ internal class YamlOutput(
     private fun emitQuotedScalar(value: String, scalarStyle: ScalarStyle) = emitScalar(value, scalarStyle)
 
     override fun encodeElement(descriptor: SerialDescriptor, index: Int): Boolean {
+        if(configuration.serializeComments) {
+            var comment = descriptor.getElementAnnotations(index)
+                .filterIsInstance<YamlComment>()
+                .firstOrNull()
+                ?.comment
+
+            if(comment != null) {
+                // Is multiline comment?
+                if(comment.indexOfAny(charArrayOf('\n', '\r')) > -1)
+                    comment = comment.trimIndent()
+
+                for (line in comment.lineSequence()) {
+                    emitter.emit(CommentEvent(CommentType.BLOCK, line, Optional.empty(), Optional.empty()))
+                }
+            }
+        }
```

Comment:
Perhaps this could be moved to a separate method to help convey the intention more clearly and help keep `encodeElement` short and clear?

Reply:
Yes, I can do it if you want